java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/SubworkValid.java

```diff
@@ -0,0 +1,191 @@
+				.map(Ausgabe_Fachdaten.class::cast)
+				.collect(Collectors
+						.groupingBy(subwork -> getNullableObject(subwork,
+								s -> s.getUntergewerkArt().getWert())));
+		groupBySubworkType.forEach((type, subworks) -> {
+			final Function<Ausgabe_Fachdaten, Map<String, String>> getSubworkGuidMap = subwork -> Map
+					.of("GUID",
+							getNullableObject(subwork,
+									s -> s.getIdentitaet().getWert())
+											.orElse(""));
+			if (type.isEmpty()) {
+				subworks.forEach(ele -> errors.add(createPlazError(ele,
+						"Die Ausgabe_Fachdaten {GUID} hat keine Untergewerk_Art",
+						getSubworkGuidMap.apply(ele))));
+				return;
+			}
+			if (subworks.size() > 1 && type
+					.get() instanceof final ENUMUntergewerkArt untergewerkArt) {
+				final String subworkTypeStr = enumTranslationService
+						.translate(untergewerkArt)
+						.getAlternative();
+				subworks.forEach(ele -> errors.add(createPlazError(ele,
+						"Die Ausgabe_Fachdaten zu Untergewerk_Art {Type} ist nicht eindeutig.",
+						Map.of("Type", subworkTypeStr))));
+			}
+		});
+		return errors;
+	}
+
+	/**
+	 * Find the {@link Ausgabe_Fachdaten} and the {@link ENUMUntergewerkArt},
+	 * which belong to this Ausgabe_Fachdaten from {@link Planung_Gruppe}
+	 * 
+	 * @param modelSession
+	 *            the {@link IModelSession}
+	 * @return the {@link Ausgabe_Fachdaten} and the {@link ENUMUntergewerkArt}
+	 */
+	private static Map<Ausgabe_Fachdaten, ENUMUntergewerkArt> getSubworkFromPlanGroup(
+			final IModelSession modelSession) {
+		final List<Planung_Gruppe> lstPlanungProjekt = getNullableObject(
+				modelSession,
+				session -> session.getPlanProSchnittstelle()
+						.getLSTPlanung()
+						.getObjektmanagement()
+						.getLSTPlanungProjekt()
+						.stream()
+						.flatMap(lstPlanung -> lstPlanung.getLSTPlanungGruppe()
+								.stream())
+						.toList()).orElse(Collections.emptyList());
+		return lstPlanungProjekt.stream()
+				.map(group -> getNullableObject(group, g -> new Pair<>(
+						g.getLSTPlanungEinzel()
+								.getIDAusgabeFachdaten()
+								.getValue(),
+						g.getPlanungGAllg().getUntergewerkArt().getWert())))
+				.filter(Optional::isPresent)
+				.map(Optional::get)
+				.collect(Collectors.toMap(Pair::getFirst, Pair::getSecond));
+	}
+
+	private PlazError createPlazError(final Ausgabe_Fachdaten subwork,
+			final String messages, final Map<String, String> data) {
+		final PlazError plazError = PlazFactory.eINSTANCE.createPlazError();
+		plazError.setObject(subwork);
+		plazError.setType(checkType());
+		plazError.setMessage(
+				StringSubstitutor.replace(messages, data, "{", "}"));
+		return plazError;
+	}
+
+	@Override
+	public String checkType() {
+		return "Untergewerk Art";
+	}
+
+	@Override
+	public String getDescription() {
+		return "Jedes Untergewerk hat maximal eine Ausgabe_Fachdaten";
+	}
+
+	@Override
+	public String getGeneralErrMsg() {
+		return "Die Ausgabe_Fachdaten je Untergewerk sind nicht eindeutig";
```

Comment:
@TruongQuangSB And then here similarly

```suggestion
		return "Die Untergewerke sind entweder nicht eindeutig oder passen nicht zu der zugehörigen Planung_Gruppe";
```